templates/usr/share/keyrings/mysql-archive-keyring-2022.gpg

```diff
@@ -0,0 +1,52 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```

Comment:
I'm getting the following error from debian when I try to add new sources, which indicates that maybe these keys aren't being registered properly:

```
    permanent: W: GPG error: http://apt.newrelic.com/debian newrelic Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B31B29E5548C16BF
    permanent: E: The repository 'http://apt.newrelic.com/debian newrelic Release' is not signed.
    permanent: W: GPG error: https://deb.nodesource.com/node_12.x buster InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 1655A0AB68576280
    permanent: E: The repository 'https://deb.nodesource.com/node_12.x buster InRelease' is not signed.
    permanent: W: GPG error: http://repo.mysql.com/apt/debian buster InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 467B942D3A79BD29
    permanent: E: The repository 'http://repo.mysql.com/apt/debian buster InRelease' is not signed.
```

Note that the files *are* copied; when I cat them they match the template, but when I cat other files that were that folder (e.g. `debian-archive-buster-security-automatic.gpg` it looks like they are binary files.  Indeed the gpg keys [in the commit you referenced as inspiration](https://github.com/OpenTechStrategies/torque-devenv/pull/29/commits/6e01a7e51818ec60e538124b2bdea0c673245895) for this change the gpg keys are binary files there too.

I think you missed the last "dearmor" step described in that commit message:

`cat ansible.asc | gpg --dearmor > ansible.gpg` <-- so if you run that on the gpg files it should do the trick

Reply:
I can confirm that running:

```
mv mysql-archive-keyring-2022.gpg mysql-archive-keyring-2022.asc
cat mysql-archive-keyring-2022.asc | gpg --dearmor > mysql-archive-keyring-2022.gpg
mv newrelic-archive-keyring.gpg newrelic-archive-keyring.asc
cat newrelic-archive-keyring.asc | gpg --dearmor > newrelic-archive-keyring.gpg
mv newrelic-archive-keyring.gpg newrelic-archive-keyring.asc
cat nodesource-archive-keyring.asc | gpg --dearmor > nodesource-archive-keyring.gpg
```

Solved the problem!  (we don't need the `.asc` files I just didn't want to give destructive advice in this message)

Reply:
So this is very interesting!

I did intentionally omit the dearmor step, for two reasons: to allow us to easily compare with upstream, which all distribute armored keys; and because binary files and git are not close friends.

What's strange is that I have tested the ability to use armored keys with apt, and it's worked for me - both on my laptop running Ubuntu 20.04, and on the dev back-end EC2 instance.

The [Debian Wiki UseThirdParty article](https://wiki.debian.org/DebianRepository/UseThirdParty) has this quote:

> The reason why we avoid ASCII-armored files is that they can only be used by [SecureApt](https://wiki.debian.org/SecureApt) in version 1.4 or later (which appeared in stretch).

The stretch release is currently also known as `oldoldstable`, and here we're using buster aka `oldstable`. I was under the impression this was simply outdated advice, intended for backwards compatibility with a version we no longer care about. So I am very confused!

I would love to understand why they are working for me and are not working for you. Short of that, I think my preferred solution would be to dearmor at install time, rather than committing the binary keys - any concern about a bash `for` loop that does some ever-so-slightly-cleverness with `basename`?

Reply:
I bit the bullet and recreated my vagrant VM with the key copy command corrected, and it worked for me. I'll push my changes, and then ask for you to re-test!

Reply:
Weird -- I wonder why the armored don't work for me.  (I did run it with the key copy correction and it didn't work until I de-armored)

I'm fine with the de-armoring happening as part of the provisioning (or if it has to be done manually as an additional step so it goes, though we should be careful not to add TOO many steps); so long as `vagrant up` works I'm fine!

Reply:
Pulled the latest -- the `.gpg` extension was the issue!  Now that they are `.asc` it works just fine as is (in fact, I may update the torque devenv to use the armored values for the reasons you described now that I see why it was trouble before!)

Thanks Jason!

Reply:
Thanks for your help!